src/Libraries/Microsoft.Extensions.Http.Resilience/Polly/HttpRetryStrategyOptionsExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Shared.DiagnosticIds;
+using Microsoft.Shared.Diagnostics;
+using Polly;
+
+namespace Microsoft.Extensions.Http.Resilience;
+
+/// <summary>
+/// Extensions for <see cref="HttpRetryStrategyOptions"/>.
+/// </summary>
+[Experimental(diagnosticId: DiagnosticIds.Experiments.Resilience, UrlFormat = DiagnosticIds.UrlFormat)]
+public static class HttpRetryStrategyOptionsExtensions
+{
+#if !NET8_0_OR_GREATER
+    private static readonly HttpMethod _connect = new("CONNECT");
+    private static readonly HttpMethod _patch = new("PATCH");
+#endif
+
+    /// <summary>
+    /// Disables retry attempts for POST, PATCH, PUT, DELETE, and CONNECT HTTP methods.
+    /// </summary>
+    /// <param name="options">The retry strategy options.</param>
+    public static void DisableForUnsafeHttpMethods(this HttpRetryStrategyOptions options)
```

Comment:
Alternatively, we can return `HttpRetryStrategyOptions` to make this and other extension-methods configuring `HttpRetryStrategyOptions` "chainable". Honestly, at the moment I don't see a necessity for that, but probably in future we'll want to introduce more API and make them fluent.

I would personally leave the API as-is returning `void`. What do you think?

Reply:
This is more a question for the API Review Board. Here are the general rules: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-change-rules.md.

I _think_ that changing the return type from `void` to `T` should be an acceptable change in the future, but I may be wrong... @terrajobst @bartonjs what's your take on this?

Reply:
> I think that changing the return type from void to T should be an acceptable change in the future, but I may be wrong...

Methods in .NET are named for their full signature, including the declared return type.  So any callers from before you change it will get a MissingMethodException if they run against a library after you change it.

If you want to change it between previews that will annoy your users, but you can do it (with a breaking change notice).  Once you abandon the preview label it is what it is forever (you'd have to leave this one as-is and make some other overload that happens to return the data, but clearly can't have the same calling signature).

Reply:
Thank you, @bartonjs for clarifications.

Reply:
FWIW, in that breaking change guidelines doc, it says it's OK to change the return _value_, but that it's not OK to change the return _type_.

> Returning a value of a more derived type for a property, field, return or out value
>
> **Note, again, that the static type cannot change.** e.g. it is OK to return a string instance where an object was returned previously, but it is not OK to change the return type from object to string.

<sub>Emphasis mine</sub>

Reply:
Let's keep the methods void. I searched dotnet/extensions and dotnet/runtime repositories, and didn't find similar public Extensions API for Options class, so there is no pattern we can follow. Usually, users directly configure Options objects by setting their properties, and API we are introducing is kind of helper-methods that allow users to easily set up existing property of the RetryOptions object. I'm not sure there will be a need to make such API "fluent" in future, but if there is we can introduce new API. For now, let's keep the API simpler, i.e. return `void`.

Reply:
If nobody has objections, I'll proceed with current API.